cmd/builder/internal/builder/templates/main.go.tmpl

```diff
@@ -16,8 +23,27 @@ func main() {
 		Description: "{{ .Distribution.Description }}",
 		Version:     "{{ .Distribution.Version }}",
 	}
+	providers := []confmap.Provider{
+		{{- range .Providers}}
+		{{.Name}}.New(),
+		{{- end}}
+	}
+	set := otelcol.CollectorSettings{
+		BuildInfo: info,
+		Factories: components,
+		ConfigProviderSettings: otelcol.ConfigProviderSettings{
+			ResolverSettings: confmap.ResolverSettings{
+				Providers:  makeMapProvidersMap(providers...),
+				Converters: []confmap.Converter{
+					{{- range .Converters}}
+					{{.Name}}.New(confmap.ConverterSettings{}),
```

Comment:
The need to pass in `confmap.ConverterSettings{}` here indicates to me that we need to make some changes to the `otelcol` API. I will issue another PR to demonstrate what I think may be a good solution.

Reply:
I think if we change the APIs a bit, we can just pass the `New` function for each module and have them instantiated later on when we have a logger to pass. I think the approach in this PR will be the cleanest and most straightforward: https://github.com/open-telemetry/opentelemetry-collector/pull/9516.

Reply:
Agreed that passing the factories here makes more sense, let's get that PR merged first